Apps/W1/HybridGP/app/src/pages/GPSetAllDimensionsDialog.Page.al

```diff
@@ -0,0 +1,155 @@
+                            BlanksClearValue := Confirm(OneDimensionBlankConfirmMsg, false, 1);
+
+                        if Dimension2 = '' then
+                            BlanksClearValue := Confirm(OneDimensionBlankConfirmMsg, false, 2);
+                    end;
+
+                    ConfirmedYes := true;
+                    CurrPage.Close();
+                end;
+            }
+
+            action(ActionCancel)
+            {
+                ApplicationArea = Basic, Suite;
+                Caption = 'Cancel';
+                Image = Cancel;
+                InFooterBar = true;
+
+                trigger OnAction()
+                begin
+                    CurrPage.Close();
+                end;
+            }
+        }
+    }
+
+    trigger OnOpenPage()
+    begin
+        GenerateUniqueSegmentIndex();
+    end;
+
+    local procedure DeleteUniqueSegmentIndex()
+    var
+        GPSegmentName: Record "GP Segment Name";
+    begin
+        GPSegmentName.SetRange("Company Name", '');
+        GPSegmentName.DeleteAll();
+    end;
+
+    local procedure GenerateUniqueSegmentIndex()
+    var
+        HybridCompany: Record "Hybrid Company";
+        GPSegmentName: Record "GP Segment Name";
+        GPSegmentNameUnique: Record "GP Segment Name";
+    begin
+        DeleteUniqueSegmentIndex();
+
+        HybridCompany.SetRange(Replicate, true);
+        if HybridCompany.FindSet() then
+            repeat
+                MigratingCompanyList.Add(HybridCompany.Name);
+            until HybridCompany.Next() = 0;
+
+        GPSegmentName.SetFilter("Company Name", '<>%1', '');
+        if GPSegmentName.FindSet() then begin
+            repeat
+                if not GPSegmentNameUnique.Get(GPSegmentName."Segment Name", '') then begin
+                    if MigratingCompanyList.IndexOf(GPSegmentName."Company Name") = 0 then
+                        break;
+
+                    GPSegmentNameUnique."Company Name" := '';
+                    GPSegmentNameUnique."Segment Name" := GPSegmentName."Segment Name";
+                    GPSegmentNameUnique."Segment Number" := GPSegmentName."Segment Number";
+                    GPSegmentNameUnique.Insert();
+                end;
+            until GPSegmentName.Next() = 0;
+        end;
+    end;
+
+    procedure GetDimension1(): Text[30]
+    begin
+        exit(Dimension1);
+    end;
+
+    procedure GetDimension2(): Text[30]
+    begin
+        exit(Dimension2);
+    end;
+
+    procedure GetBlanksClearValue(): Boolean
+    begin
+        exit(BlanksClearValue);
+    end;
+
+    procedure GetConfirmedYes(): Boolean
+    begin
+        exit(ConfirmedYes);
+    end;
+
+    var
+        MigratingCompanyList: List of [Text];
+        Dimension1: Text[30];
+        Dimension2: Text[30];
+        BlanksClearValue: Boolean;
+        ConfirmedYes: Boolean;
+        GlobalDimensionsCannotBeTheSameErr: Label 'Dimension 1 and Dimension 2 cannot be the same.', Locked = true;
```

Comment:
Same comment for locked, it should be removed so it can be translated

Reply:
Updated